src/dlgProfilePreferences.cpp

```diff
@@ -258,27 +258,81 @@ dlgProfilePreferences::dlgProfilePreferences(QWidget* pF, Host* pHost)
 
     comboBox_guiLanguage->clear();
     for (auto& code : pMudlet->getAvailableTranslationCodes()) {
-        auto& translatedName = pMudlet->mLanguageCodeMap.value(code).first;
-        int translatedPc = pMudlet->mLanguageCodeMap.value(code).second;
-        comboBox_guiLanguage->addItem(translatedName, code);
-        if (translatedPc >= pMudlet->mTranslationStar) {
-            comboBox_guiLanguage->setItemIcon(comboBox_guiLanguage->count()-1, QIcon(":/icons/rating.png"));
+        auto& translation = pMudlet->mTranslationsMap.value(code);
+        auto& nativeName = translation.getNativeName();
+        if (translation.getIsFromResourceFile()) {
+            auto& translatedPc = translation.getTranslatedPercentage();
+            if (translatedPc >= pMudlet->mTranslationGoldStar) {
+                comboBox_guiLanguage->addItem(QIcon(":/icons/rating.png"),
+                                              nativeName,
+                                              code);
+            } else if (translatedPc >= pMudlet->mTranslationSilverStar) {
+                comboBox_guiLanguage->addItem(QIcon(":/icons/rating_silver.png"),
+                                              tr("%1 (%2% done)",
+                                                 // Intentional argument to separate arguments
+                                                 "%1 is the (not-translated so users of the language can read it!) language name, %2 is percentage done, which is high enough to show a silver star!")
+                                              .arg(nativeName, QString::number(translatedPc)),
+                                              code);
+            } else {
+                // This will also be used if the percentage is set to zero
+                // because it was not found in the translation statistics file
+                // during compilation even though the Mudlet translation is in
+                // the resources file:
+                comboBox_guiLanguage->addItem(QIcon(),
+                                              tr("%1 (%2% done)",
+                                                 // Intentional argument to separate arguments
+                                                 "%1 is the (not-translated so users of the language can read it!) language name, %2 is percentage done, which is not high enough to show a gold or silver star!")
```

Comment:
This string `"%1 (%2% done)` will now show up twice on Crowdin, because both ocurrences in code have a (very slightly) different comment. Please use same comment instead.

Reply:
I see where you are coming from - but I thought the purpose of these comments were to distinguish between otherwise identical source texts - the Qt Docs do say:

> `[static] QString QObject::tr(const char *sourceText, const char *disambiguation = Q_OBJECT, int n = Q_OBJECT)`
> ...
> If the same *sourceText* is used in different roles within the same context, an additional identifying string may be passed in *disambiguation* (0 by default). In Qt 4.4 and earlier, this was the preferred way to pass comments to translators.

OTOH in this situation the roles are linguistically identical - so there is no merit in them being different.

So, yeah, I'll make them the same.

Reply:
Due to some reversions this is no longer an issue!